plugins/chat/app/models/concerns/chat/thread_cache.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module Chat
+  module ThreadCache
+    extend ActiveSupport::Concern
+
+    class_methods do
+      def replies_count_cache_updated_at_redis_key(id)
+        "chat_thread:replies_count_cache_updated_at:#{id}"
+      end
+
+      def replies_count_cache_redis_key(id)
+        "chat_thread:replies_count_cache:#{id}"
+      end
+
+      def clear_caches!(ids = nil)
+        return Discourse.redis.delete_prefixed("chat_thread:") if ids.blank?
+        ids = Array.wrap(ids)
+
+        keys_to_delete =
+          ids
+            .map do |id|
+              [replies_count_cache_redis_key(id), replies_count_cache_updated_at_redis_key(id)]
+            end
+            .flatten
+        Discourse.redis.del(keys_to_delete)
+      end
+    end
+
+    def replies_count_cache_recently_updated?
+      replies_count_cache_updated_at.after?(5.minutes.ago)
+    end
+
+    def replies_count_cache_updated_at
+      Time.at(
+        Discourse.redis.get(Chat::Thread.replies_count_cache_updated_at_redis_key(self.id)).to_i,
+        in: Time.zone,
+      )
+    end
+
+    def replies_count_cache
+      redis_cache = Discourse.redis.get(Chat::Thread.replies_count_cache_redis_key(self.id))&.to_i
+      if redis_cache.present? && redis_cache != self.replies_count
+        redis_cache
+      else
+        self.replies_count
+      end
+    end
+
+    def set_replies_count_cache(value, update_db: false)
+      self.update!(replies_count: value) if update_db
+      Discourse.redis.setex(
+        Chat::Thread.replies_count_cache_redis_key(self.id),
+        5.minutes.from_now.to_i,
+        value,
+      )
+      Jobs.enqueue_in(3.seconds, Jobs::Chat::UpdateThreadReplyCount, thread_id: self.id)
```

Comment:
What is the meaning of `3.seconds` here ? (as in, why not 5 or 2 ?)

Reply:
Just what I used for the user_count for channels, we can make it whatever we want really. Might actually change it to 5 seconds to space them out more.